torch/distributed/fsdp/sharded_grad_scaler.py

```diff
@@ -281,27 +281,41 @@ def unscale_(self, optimizer: torch.optim.Optimizer) -> None:
         # Synchronize the detected inf across the ranks
         optimizer_state = self._per_optimizer_states[id(optimizer)]
         future_handles = []
-
-        for v in optimizer_state["found_inf_per_device"].values():
-            if v.device.type == "cpu":
-                v_on_cuda = v.cuda()
+        found_inf_on_cudas = []
+
+        for found_inf in optimizer_state["found_inf_per_device"].values():
+            if found_inf.device.type == "cpu":
```

Comment:
When is `found_inf.device.type != "cpu":`? Only when CPU offloading?

Reply:
found_inf.device.type == "cpu" is decided by cpu offloading only. found_inf.device is decided by `param.grad.device` from  `optimizer.param_groups.group["params"]` in current implementation